x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/X509AuthenticationToken.java

```diff
@@ -24,7 +25,7 @@ public X509AuthenticationToken(X509Certificate[] certificates) {
 
     public X509AuthenticationToken(X509Certificate[] certificates, boolean isDelegated) {
         this.credentials = Objects.requireNonNull(certificates);
-        if (false == CertParsingUtils.isOrderedCertificateChain(certificates)) {
+        if (false == CertParsingUtils.isOrderedCertificateChain(Arrays.asList(certificates))) {
```

Comment:
Unfortunately `trustManager.checkClientTrusted` inside the `PKiRealm` still deals with arrays instead of lists, so we would be shifting the conversion to other places. I prefer to keep the lists/arrays as they are (or have been proposed in this PR). Probably not ideal, but I prefer to minimize the changes noise in this case; unless someone thinks otherwise, ofc.